corehq/motech/repeaters/models.py

```diff
@@ -310,6 +310,10 @@ def _optionvalue_fields(self):
             if isinstance(attr_tuple[1], OptionValue)
         ]
 
+    @property
+    def get_id(self):
```

Comment:
I thought `get_id` was typically a synonym for `_id`, which in SQL is normally `id` or `pk`. Is that not the case with these models?

Rather than bringing this awful couchdbkit property over to SQL models, would it be difficult to change the places that reference `.get_id` to use `.repeater_id` instead? Having multiple names for the same property gets confusing and smells of legacy code.

Might be made it a bit easier by adding a `repeater_id` property to the old Couch class(es) for compatibility, although avoid if possible, especially if there are other means to know that the code is operating on SQL models and not Couch docs. The down-side to this is jsonobject will persist this new property in Couch when saving the document. There may be other ways to get around that. Let me know if it becomes an issue.

In general, add compatibility helpers to the old Couch models that will eventually be deleted, and try to keep all new code clean and following best practices so we end up in a better state than we were before.

Reply:
Generally +1 on this though I would not want to see 'repeater_id' being used in code that is meant to be model agnostic.

I see that `repeater_id` is the 'couchd ID'. If we ever plan to remove that we should stick with the SQL primary key.

Reply:
> I would not want to see 'repeater_id' being used in code that is meant to be model agnostic.

@snopoke can you give an example of what you mean by this?

Reply:
What I mean is that `get_id` is a generic method that works for all couch models, like `model.pk` works for SQL models but 'repeater_id' is specific to repeaters.

The code that is causing this change should ideally be as model agnostic as possible.

Reply:
> I see that repeater_id is the 'couchd ID'. If we ever plan to remove that we should stick with the SQL primary key.

I think generally when it's given a name other than `couch_id` it is intended to remain indefinitely. I wonder if we could have avoided this by using a UUID primary key (similar to [what was done with lookup tables](https://github.com/dimagi/commcare-hq/blob/311a17bea8be56e7cb5edc1d37b87775af6c299c/corehq/apps/fixtures/models.py#L63))?

Note that there are some gotchas to work around with an transition to a UUID primary key.

https://github.com/dimagi/commcare-hq/blob/311a17bea8be56e7cb5edc1d37b87775af6c299c/corehq/apps/fixtures/models.py#L75-L79

Reply:
> Rather than bringing this awful couchdbkit property over to SQL models, would it be difficult to change the places that reference .get_id to use .repeater_id instead? Having multiple names for the same property gets confusing and smells of legacy code.

It is fairly easy to remove `get_id`. I was confused about this one and I looked for CommCareCase SQL model which implements a `get_id` method so I went ahead with this method. I can remove it from the model.

> Generally +1 on this though I would not want to see 'repeater_id' being used in code that is meant to be model agnostic.
I see that repeater_id is the 'couchd ID'. If we ever plan to remove that we should stick with the SQL primary key.

The reason we went ahead with keeping `repeater_id` was as it was used/referenced in many places and in data(specifically repeat records) and in some URLs. So we decided to keep using `repeater_id` and not using the SQL pk.
But I see your point where it can cause issues in generalizing the code.

Reply:
> It is fairly easy to remove `get_id`.

:+1: 

> CommCareCase SQL model ... implements a `get_id` method

I think that exists from before we started migrating data from Couch to SQL, and I think it would be better not to continue that pattern.